### ISSUE TITLE
Allow `defer` to know if the task succeded or failed

### ISSUE THREAD

**Issue by andreynering** [go-task/task#1484]
Originally asked in #1472 by @your-diary.

A possible way to allow this is to expose one/some variables to `defer`. One possibility is to expose a `DEFER_EXIT_CODE` var which would contain `0` for success or the exit code of the task that failed.

Perhaps we could also have `DEFER_SUCCESS` (true/false) and `DEFER_FAIL` (true/false) to make a simpler check easier? (Not sure, `DEFER_EXIT_CODE` is probably enough).

**Comment by dorimon-1** [comment]
@andreynering Could you explain a bit deeper how you would the yml to be?
I thought about 
```
  task-2:
    - defer_failure: echo Failure!
    - defer_failure: { task: "task-1", vars: { PARAM: "failure" } }
    - defer_success: echo Success!
    - defer_success: { task: "task-1", vars: { PARAM: "success" } }
    - echo 'cmd ran'
    - exit 1
```

**Comment by your-diary** [comment]
@dorimon-1 
One can ignore my original suggestion. It is just an example.

As for your suggestion, if we add `defer_success` and `defer_failure` while keeping `defer` keyword, we can make it backward-compatible.

**Comment by KJone1** [comment]
Exactly what I have been searching for.
 
I think @dorimon-1 idea looks neat and allows for backward compatibility with regular defer. 
Also having env var for defer status code could be helpful for some ppl.
If someone could implement this it would be much appreciated 😉.

**Comment by andreynering** [comment]
This was just implemented on #1762.
